apps/examples/wifi_manager/wm_test/wm_test_dns.c

```diff
@@ -0,0 +1,119 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License\n");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+#include <netdb.h>
+#include <semaphore.h>
+#include <wifi_manager/wifi_manager.h>
+#include <stress_tool/st_perf.h>
+#include "lwip/lwipopts.h"
+#include "wm_test.h"
+#include "wm_test_utils.h"
+#include "wm_test_network.h"
+#include "wm_test_log.h"
+
+#define TAG "[WT]"
+
+static sem_t g_wm_sem;
+
+static char *g_host_list[] = {
+	"www.google.com",
+	"www.facebook.com",
+	"www.naver.com",
+	"www.daum.net",
+};
+
+TESTCASE_SETUP(simple_tc)
+{
+	/*  setup DNS server 8.8.8.8 */
+	struct sockaddr_in dns_addr;
+	dns_addr.sin_family = AF_INET;
+	inet_pton(AF_INET, "8.8.8.8", (void *)&dns_addr.sin_addr);
+	ST_ASSERT_EQ(0, netlib_setdnsserver((struct sockaddr *)&dns_addr, -1));
+} END_TESTCASE
+
+TESTCASE_TEARDOWN(simple_tc)
+{
+} END_TESTCASE
+
+static void _print_result(char *hostname, struct addrinfo *local_res)
+{
+	struct addrinfo *ainfo = NULL;
+	WT_LOGP(TAG, "Host: %s\n", hostname);
+	for (ainfo = local_res; ainfo != NULL; ainfo = ainfo->ai_next) {
+		WT_LOGP(TAG, "family %s\n", ((ainfo->ai_family == AF_INET) ? "AF_INET" : "Not IPv4"));
+		if (ainfo->ai_family != AF_INET) {
+			WT_LOGP(TAG, "not interesting type %d\n", ainfo->ai_family);
+			continue;
+		}
+		struct sockaddr_in *dst = (struct sockaddr_in *)ainfo->ai_addr;
+		struct in_addr info_addr = dst->sin_addr;
+		uint8_t addr[4] = {0,};
+		addr[0] = info_addr.s_addr & 0xFF;
+		addr[1] = (info_addr.s_addr >> 8) & 0xFF;
+		addr[2] = (info_addr.s_addr >> 16) & 0xFF;
+		addr[3] = (info_addr.s_addr >> 24) & 0xFF;
+		WT_LOGP(TAG, "%d.%d.%d.%d\n", addr[0], addr[1], addr[2], addr[3]);
+	}
+	WT_LOGP(TAG, "\n");
+}
+
+START_TEST_F(dns_queue)
+{
+	struct addrinfo hints, *local_res;
+	int arr_size = sizeof(g_host_list) / sizeof(char *);
+	memset(&hints, 0, sizeof(hints));
```

Comment:
I think `sizeof(struct addrinfo)` is better.

Reply:
Thanks I'll modify it